File: leetcode/[27]RemoveElement.js
// SOLUTION 1
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
*/
var removeElement = function(nums, val) {
    const newNums = nums.filter((el) => el !== val);
    for (let i = 0; i < newNums.length; i++) {
        nums[i] = newNums[i];
    }
    
    console.log(newNums);
    console.log(nums);
    return nums;
};

// SOLUTION 2
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let index = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] != val) {
            nums[index] = nums[i];
            index++;
        }
    }
    return index;
}; 

File: leetcode/[27]RemoveElement.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        nums[:] = [num for num in nums if num != val]
        return len(nums)
                        
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        slow = 0
        for fast in range(len(nums)):
            if nums[fast] != val:
                nums[slow] = nums[fast]
                slow += 1
        return slow